static/styles/zulip.scss

```diff
@@ -433,6 +433,31 @@ li,
     position: absolute !important;
 }
 
+/* Headings */
+
+.message_content h1,
+.message_content h2,
+.message_content h3,
+.message_content h4,
+.message_content h5,
+.message_content h6 {
+    font-size: 14px;
+    font-weight: 600;
+    margin: 0;
+    line-height: 30px;
+}
+
+.message_content h1::after,
+.message_content h2::after,
+.message_content h3::after,
+.message_content h4::after,
+.message_content h5::after,
+.message_content h6::after, {
+    content: '';
+    display: block;
+    border: 0.5px solid hsla(0, 0%, 0%, 0.1);
+}
```

Comment:
These should all go inside `.rendered_markdown`, which is our master stylesheet for the markdown system.